client/pages/signup.jsx

```diff
@@ -8,58 +8,97 @@ export const addUserMutation = gql`
-              id="password"
-              placeholder="Password"
-              value={password}
-              onChange={event => {
-                setPassword(event.target.value);
-              }}
-              required
-            />
+            <div className="form-group">
+              <label htmlFor="email" className="float-left required">
+                Email address
+              </label>
+              <input
+                type="email"
+                className="form-control"
+                id="email"
+                placeholder="Enter your email"
+                value={email}
+                onChange={event => {
+                  setEmail(event.target.value);
+                }}
+                autoFocus
+                required
+              />
+              <small
+                id="emailHelp"
+                className="form-text text-muted float-left mb-3"
+              >
+                We&apos;ll never share your email with anyone else.
+              </small>
+              <br />
+            </div>
+            <div className="form-group">
+              <label htmlFor="password" className="float-left required">
+                Password
+              </label>
+              <input
+                type="password"
+                className="form-control"
+                id="password"
+                placeholder="Enter your password"
+                value={password}
+                onChange={event => {
+                  setPassword(event.target.value);
+                }}
+                required
+              />
+              <small
+                id="emailHelp"
+                className="form-text text-muted float-left mb-3 d-inline-block"
+              >
+                Must contain at least 6 alphanumeric characters.
+              </small>
+              <br />
+            </div>
+
             <button
               type="submit"
               className="btn btn-primary float-left"
               onClick={async event => {
+                setErrorMessage("");
+                setSuccessMessage("");
                 event.preventDefault();
-                await client.mutate({
+                const { data } = await client.mutate({
                   mutation: addUserMutation,
                   variables: { email, password }
                 });
+                if (data.addUser.email === "empty") {
+                  setErrorMessage("You must enter a valid email address");
+                } else if (data.addUser.email === "taken") {
+                  setErrorMessage("That email address is already in use.");
+                } else {
+                  setSuccessMessage("Your account has been created!");
+                }
                 setEmail("");
                 setPassword("");
               }}
             >
               Sign in
             </button>
           </form>
+          <p className="text-center text-danger">{errorMessage}</p>
```

Comment:
I think you should not render this tag when `errorMessage` is empty.
The same goes for the line below.

Reply:
I'll see about modifying this.

Reply:
Changed to code to conditionally render the `<p>` tag with a message in the following way:
```
          {errorMessage.length ? (
              <p className="text-center text-danger">{errorMessage}</p>
          ) : ( successMessage.length > 0 &&
              <p className="text-center text-success">{successMessage}</p>
          )}
```

Reply:
I'll go ahead and merge this. Thanks, @allanapplebee  and @hitochan777 . 
@hitochan777 I believe I've done what you requested, so I'll click 'dismiss review' in order to merge this, but of course that's just a formality :) 
If you'd like me to handle adjust the code some more, let me know.